src/main/java/org/mvel2/execution/ExecutionArrayList.java

```
@@ -116,13 +165,134 @@ public long memorySize() {
+        newList.sort(asc);
+        return newList;
+    }
+
+    public void reversed() {
+        Collections.reverse(this);
+    }
+
+    public List toReversed() {
+        ExecutionArrayList newList = this.slice();
+        newList.reversed();
+        return newList;
+    }
+
+    public List concat(Collection c) {
+        ExecutionArrayList newList = this.slice();
+        newList.addAll(c);
+        return newList;
+    }
+
+    public List splice(int start) {
+        return this.splice(start, null);
+    }
+
+    public List splice(Object st, Object delCount, E... items) {
+        int start = Integer.parseInt(String.valueOf(st));
+        start = start < -this.size() ? 0 : start >= this.size() ? this.size() : start < 0 ? start + this.size() : start;
+        int deleteCount = delCount == null ? this.size() - start : Integer.parseInt(String.valueOf(delCount));
+        deleteCount = start == this.size() ? 0 : Math.max(deleteCount, 0);
+        List<E> removed = new ArrayList<>();
+        int deleteIdx = deleteCount == 0 ? this.size() : start;
+        AtomicInteger insertIdx = new AtomicInteger(start);
+        while (deleteCount > 0) {
+            removed.add(this.remove(deleteIdx));
+            deleteCount--;
+        }
+        for (E e : items) {
+            this.add(insertIdx.getAndIncrement(), e);
+        }
+        return new ExecutionArrayList<>(removed, this.executionContext);
+    }
+
+    public List toSpliced(int start) {
+        return this.toSpliced(start, null);
+    }
+
+    public List toSpliced(Object st, Object delCount, E... items) {
+        ExecutionArrayList newList = this.slice();
+        newList.splice(st, delCount, items);
+        return newList;
+    }
+
+    public List with(Object ind, E items) {
+        int index = Integer.parseInt(String.valueOf(ind));
+        if (index >= this.size() || index < -this.size()) {
+            this.add(index, items);
+            return null;
```

throw error instead of null:
If the index after normalization is out of bounds, a [RangeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError) is thrown.

---

Ok